meson.build

```diff
@@ -242,10 +242,28 @@ if build_daemon
     warning('Polkit is disabled, the daemon will allow ALL client actions')
   endif
   udevdir = get_option('udevdir')
-  if udevdir == ''
+  if udevdir == '' and host_machine.system() == 'linux'
     udev = dependency('udev')
     udevdir = udev.get_pkgconfig_variable('udevdir')
   endif
+  # Check if machine-id exists
+  fs = import('fs')
+  machine_id_paths = [
+    '/var/local/lib/dbus/machine-id',
+    '/var/local/etc/machine-id"',
```

Comment:
these paths should not be hardcoded, but should instead reference the arguments like previously.  build a path using them.